src/dotnet/commands/dotnet-migrate/MigrateCommand.cs

```diff
@@ -41,18 +54,81 @@ public int Execute()
 
             EnsureNotNull(sdkVersion, "Null Sdk Version");
 
+            MigrationReport migrationReport = null;
+
             foreach (var project in projectsToMigrate)
             {
                 Console.WriteLine($"Migrating project {project}..");
                 var projectDirectory = Path.GetDirectoryName(project);
                 var outputDirectory = projectDirectory;
                 var migrationSettings = new MigrationSettings(projectDirectory, outputDirectory, sdkVersion, msBuildTemplate, _xprojFilePath);
-                new ProjectMigrator().Migrate(migrationSettings, _skipProjectReferences);
+                var projectMigrationReport = new ProjectMigrator().Migrate(migrationSettings, _skipProjectReferences);
+
+                if (migrationReport == null)
+                {
+                    migrationReport = projectMigrationReport;
+                }
+                else
+                {
+                    migrationReport = migrationReport.MergeReport(projectMigrationReport);
+                }
             }
 
+            WriteReport(migrationReport);
+
             return 0;
         }
 
+        private void WriteReport(MigrationReport migrationReport)
+        {
+            using (var outputTextWriter = GetReportOutputTextWriter())
+            {
+                outputTextWriter.Write(GetReportContent(migrationReport));
+            }
+        }
+
+        private string GetReportContent(MigrationReport migrationReport)
+        {
+            if (_reportFormatJson)
+            {
+                return Newtonsoft.Json.JsonConvert.SerializeObject(migrationReport);
+            }
+
+            StringBuilder sb = new StringBuilder();
+
+            foreach (var projectMigrationReport in migrationReport.ProjectMigrationReports)
+            {
+                if (projectMigrationReport.Errors.Any())
+                {
+                    sb.AppendLine($"Project {projectMigrationReport.ProjectName} migration failed ({projectMigrationReport.ProjectDirectory})");
+                    sb.AppendLine(string.Join("\n", projectMigrationReport.Errors.Select(e => e.GetFormattedErrorMessage())));
+                    sb.AppendLine();
+                }
+                else
+                {
+                    sb.AppendLine($"Project {projectMigrationReport.ProjectName} migration succeeded ({projectMigrationReport.ProjectDirectory})");
```

Comment:
Yeah, I'm going to make a change to always bring the report even if you specify a file and then that should be doable. The issue I ran into was with using the TextWriter rather than Console to write.